src/DynamoCore/Library/FunctionDescriptor.cs

```diff
@@ -568,4 +569,217 @@ private string GetRootCategory()
             return string.IsNullOrEmpty(Namespace) ? filename : filename + "." + Namespace;
         }
     }
+
+    internal class ReflectionFunctionDescriptor : FunctionDescriptor
+    {
+        private readonly LibraryCustomization libraryCustomization;
+        private readonly Assembly reflectionAsm;
+
+        public new string Category
+        {
+            get
+            {
+                var categoryBuf = new StringBuilder();
+                categoryBuf.Append(GetRootCategory());
+
+                //if this is not BuiltIn function search NodeCategoryAttribute for it
+                if (ClassName != null)
+                {
+
+                    if (reflectionAsm != null && reflectionAsm.GetType(ClassName) != null)
```

Comment:
```suggestion
                    if (reflectionAsm != null && reflectionAsm.GetType(ClassName) is Type type)
```
This should give you the required type in an atomic operation and remove the need for the `where` query below